### DISCUSSION TITLE
🎧 Kaizen 8 - 24 improvements & a lot more - Nov. 23, 2022

### DISCUSSION THREAD

**Discussion by gerhard** [thechangelog/changelog.com#433] [Kaizen]
[![shields.io](https://img.shields.io/badge/Last%20updated%20on%20Nov.%2023%2C%202022-PUBLISHED-blue?style=for-the-badge)](https://shipit.show/80)

[This is now available as shipit.show/80!](https://changelog.com/shipit/80) See all improvements in this 🐙 GitHub repository: [Kaizen 7 ... 8](https://github.com/thechangelog/changelog.com/compare/kaizen.7...kaizen.8)

<a href="https://changelog.com/shipit/80"><img width="1440" alt="image" src="https://user-images.githubusercontent.com/3342/212471679-f511adff-766a-4ab8-a2a0-454d6afd33bc.png"></a>

---

For the last Kaizen episode of 2022 - https://shipit.show/80 🎧 Nov. 23, 2022 - I want us to improve the conversation with our listeners. Rather than reference some PRs & talk about the changes that we did in the 10 weeks since the previous Kaizen, I want us to create an open-ended discussion (like this one) which gives everyone the opportunity to participate _as the improvements are happening_. Instead of waiting to hear what happened, if you are interested and are able to make the time, you can be part of it. You can start by leaving a comment below, or simply adding a reaction to this post 👍 👎 😄  🎉 😕 ❤️ 🚀 👀

<img align="right" width="294" alt="image" src="https://user-images.githubusercontent.com/3342/200187842-6366ddf0-2170-4ebe-a86b-688ea63f0ceb.png">

On top of allowing others to participate, we (+@jerodsanto +@adamstac) can experiment with keeping the episode notes in public (yes, directly in this discussion), rather than in a private doc. By the way, we have been using [HackMD](https://hackmd.io) for this purpose for all previous Kaizens. An outline from 📻 [previous Kaizen 7](https://shipit.show/70) 👉

Before I wrap up this intro, I want to mention the [Arbitrary deadlines are actually awesome](https://changelog.com/posts/arbitrary-deadlines-are-actually-awesome) article that @jerodsanto posted a month ago. As it turns out, that was a follow-up to a blog post by @lucasfcosta. This is the paragraph that seeded the idea of these Kaizen discussions:

> Every tenth episode of [Ship It!](https://shipit.show/) we talk about the work we’re doing here at [Changelog](https://changelog.com/) to make things better. That means I know that every two and a half months, I’m gonna go on a podcast and talk about what I accomplished. So when that sucker is coming up my brain is like, “We’re shipping stuff!”
>
> Jerod Santo - Oct. 19, 2022

The last thing which I want to mention is that, while I have a good feeling about this initiative and want to try it out, as with any experiment, this might be just a one off. I don't have any expectations beyond giving it a go and seeing what happens - sound familiar? I am especially curious how easy & straightforward this will be in practice for @jerodsanto & @adamstac. If you are a committed Kaizen fan and want to share your perspective, I look forward to reading your comment(s) below.

## What are the 🎧 [Kaizen 8](https://shipit.show/80) highlights?

Before we unpack the highlights, I want to start off with [something that I have not done since 2019](https://changelog.com/posts/the-new-changelog-setup-for-2019): a diagram of our setup. I suspect that @jerodsanto & @adamstac will appreciate being able to visualise all the components of our current setup 😎

```mermaid
graph TB
    classDef link stroke:#59b287,stroke-width:3px;
    %% Code & assets
    subgraph GitHub
        repo{{ fab:fa-github thechangelog/changelog.com }}:::link
        click repo "https://github.com/thechangelog/changelog.com"
        cicd[/ fa:fa-circle-check GitHub Action - Ship It \]:::link
        click cicd "https://github.com/thechangelog/changelog.com/actions/workflows/ship_it.yml"
        automation[\ fa:fa-space-shuttle dagger v0.1 /]:::link
        click automation "https://github.com/thechangelog/changelog.com/blob/master/2021/dagger/prod_image/main.cue"
    end
    
    registry(( fab:fa-docker Docker Hub )):::link
    click registry "https://hub.docker.com/r/thechangelog/changelog.com/tags"
    chat(( fab:fa-slack Slack )):::link
    click chat "https://changelog.slack.com/archives/C03SA8VE2"

    repo -.- |.github/workflows/ship_it.yml| cicd
    repo -.- |2021/dagger/prod_image/main.cue| automation
    
    cicd --> |runs| automation ---> |wireguard| container --> |app image| registry -.-> |app image| app
    cicd ---> |flyctl deploy| app
    cicd ----> |success #dev| chat

    repo -.- |2022.fly config| app
    repo -.- |2022.fly/docker config| container

    repo -.-> |make runtime-image| registry -.-> |runtime image| automation

    %% PaaS - https://fly.io/dashboard/changelog
    subgraph Fly.io
        proxy{fa:fa-globe Proxy}
        proxy ====> |https| app

        subgraph IAD
            container([ fab:fa-docker Docker Engine 2022-03-13 ]):::link
            click container "https://fly.io/apps/docker-2022-06-13"
            app(( fab:fa-phoenix-framework App changelog-2022-03-13.fly.dev )):::link
            style app fill:#488969;
            click app "https://fly.io/apps/changelog-2022-03-13"
            dbw([ fa:fa-database PostgreSQL Leader 2022-03-12 ]):::link
            click dbw "https://fly.io/apps/postgres-2022-03-12"
            dbr1([ fa:fa-database PostgreSQL Replica 2022-03-12 ])

            app <==> |pgsql| dbw
            dbw -.-> |replication| dbr1
        end

        metricsdb([ fa:fa-chart-line Prometheus ])
        metrics[ fa:fa-columns Grafana fly-metrics.net ]:::link
        click metrics "https://fly-metrics.net"
        metrics --- |promql| metricsdb
        metricsdb -..- |metrics| app
        metricsdb -..- |metrics| dbw
        metricsdb -..- |metrics| container
    end

    secrets(( fa:fa-key 1Password )):::link
    click secrets "https://changelog.1password.com/"
    secrets -.-> |secrets| app
        secrets -.-> |secrets| cicd

        
    %% Exceptions
    exceptions(( fa:fa-car-crash Sentry )):::link
    click exceptions "https://sentry.io/organizations/changelog-media/issues/?project=5668962"
    app -..-> |exceptions| exceptions

    %% CDN - https://manage.fastly.com/configure/services/7gKbcKSKGDyqU7IuDr43eG
    subgraph Fastly
        apex[ changelog.com ]:::link
        click apex "https://changelog.com"
        
        subgraph Ashburn
            cdn[ cdn.changelog.com ]
        end
    end
    observability(( fa:fa-bug Honeycomb )):::link
    click observability "https://ui.honeycomb.io/changelog/boards"
    apex -.-> |logs| observability
    apex ==> |https| proxy
    subgraph AWS.S3
        subgraph us-east-1
            assets[ fab:fa-aws changelog-assets.s3.amazonaws.com ]
        end
    end
    cdn ==> |https| assets

    %% Monitoring
    monitoring(( fa:fa-table-tennis Pingdom )):::link
    click monitoring "https://my.pingdom.com/app/newchecks/checks"
    monitoring -.-> |monitors| apex
    monitoring -.-> |monitors| cdn
    monitoring -.-> |monitors| app
```

👆 All those 🟩 green elements are links - click them and see what happens!

> Ever since I spoke with @kelseyhightower on 🔈 [shipit.show/44](https://shipit.show/44), I really wanted to document how everything works. While this is just [a mermaid flowchart](https://mermaid-js.github.io/mermaid/#/flowchart), it is a step in that direction (and hopefully worth a thousand words). Click on the ↔ to expand the diagram for the best viewing experience - it looks the part on 2880 x 1620.

Before we move on, take a moment to consider just how better having these diagrams is for visualising improvements. Did I mention that they are all code? If you take one thing away from the new Kaizen experiment, this would be it: **Put some 🧜‍♀️ https://mermaid.live in your GitHub markdown files**

OK, so what are the other highlights?

### 1/3. Kaizen tags

I thought that it would be a good idea to tag all commits in this repository which matched the publish date of every previous Kaizen episode. You will find them all here: https://github.com/thechangelog/changelog.com/tags

My favourite part is that now we can all see what changed in our changelog.com setup between the Kaizens:
- [Kaizen 1 ... 2](https://github.com/thechangelog/changelog.com/compare/kaizen.1...kaizen.2)
- [Kaizen 2 ... 3](https://github.com/thechangelog/changelog.com/compare/kaizen.2...kaizen.3)
- [Kaizen 3 ... 4](https://github.com/thechangelog/changelog.com/compare/kaizen.3...kaizen.4)
- [Kaizen 4 ... 5](https://github.com/thechangelog/changelog.com/compare/kaizen.4...kaizen.5)
- [Kaizen 5 ... 6](https://github.com/thechangelog/changelog.com/compare/kaizen.5...kaizen.6)
- [Kaizen 6 ... 7](https://github.com/thechangelog/changelog.com/compare/kaizen.6...kaizen.7)

Which brings us to today, a few weeks before Kaizen 8: [Kaizen 7 ... today, Nov. 13](https://github.com/thechangelog/changelog.com/compare/kaizen.7...69d18bcaaa4e0f39a285a7ea998944641b904026). As you can see, @jerodsanto was rather busy 🐰 . I know what you're thinking: could we have a higher-level view of what actually changed? Do you mean something like a... changelog 😁

### 2/3. Issues & PRs

These are the issues that we have closed since **Kaizen 7**:

- https://github.com/thechangelog/changelog.com/issues/392
- https://github.com/thechangelog/changelog.com/issues/425
- https://github.com/thechangelog/changelog.com/issues/427
- https://github.com/thechangelog/changelog.com/issues/429
- https://github.com/thechangelog/changelog.com/issues/432

And here are all the improvements that shipped as PRs since **Kaizen 7**:

- https://github.com/thechangelog/changelog.com/pull/415 🙏 Thank you @gilacost 
- https://github.com/thechangelog/changelog.com/pull/426
- https://github.com/thechangelog/changelog.com/pull/428 🙏 Thank you @Nezteb 
  - two follow-ups for @jerodsanto https://github.com/thechangelog/changelog.com/pull/428#issuecomment-1312784093
- https://github.com/thechangelog/changelog.com/pull/431
- https://github.com/thechangelog/changelog.com/pull/434
- https://github.com/thechangelog/changelog.com/pull/436

There is also a bunch of changes that we - it's the royal **we**, this is all @jerodsanto - did as small improvements shipped straight into production (no PR):

1. [[admin] Queue a AudioUpdater after episodes get updated](https://github.com/thechangelog/changelog.com/commit/2e2327ed06f68cb03ebc3a5db9e5ca9e2fa2060c)
2. [Replace Google reCAPTCHA with Cloudflare Turnstile](https://github.com/thechangelog/changelog.com/commit/35edb3285840a08f98241fe17145761bc542156f)
3. [Add a field to distinguish news publications from platforms](https://github.com/thechangelog/changelog.com/commit/f63d3f255f9f49fc605535ee55e9951360d6bae2) + https://github.com/thechangelog/changelog.com/commit/ee974d4b9630975cf78adbdb84dc92cbe8005b2a
4. [Redirect Medium posts to Scribe instead](https://github.com/thechangelog/changelog.com/commit/ff37593b70077e114a599c4a2c420c8def0d8e5d)
5. [[admin] Add a button for copying chapter data to ++ file](https://github.com/thechangelog/changelog.com/commit/d6a45522cc2c7aa78f5ef1f57f2d12249df4662d)
6. [Ensure chapter links and images have full URL](https://github.com/thechangelog/changelog.com/commit/f4a5fcc8199505a67cd17cce533e2d4100da8ac3) + https://github.com/thechangelog/changelog.com/commit/5cf8cea847b908ee0e3b746d53abb72502d7c34b
7. [Only show author avatar on news items if one has been uploaded](https://github.com/thechangelog/changelog.com/commit/272d542fdd1fa68cdd4b80a063fcdf0cfc2590d1)
8. [[admin] Add image URL fields to chapters form](https://github.com/thechangelog/changelog.com/commit/3d485c9f00c3c0323888bd0ca66f44fb7c276463)
9. [Update mix deps](https://github.com/thechangelog/changelog.com/commit/126a20a3599bfbde690281f52fa298aa99172b73)
10. [Improve person-related data deletion rules](https://github.com/thechangelog/changelog.com/commit/51cee276e4154cdbe3c30bb48cae79f49365b355)
11. [Improve how we track visited news links](https://github.com/thechangelog/changelog.com/commit/3760267067d75c88514c3f5e848883d5bacd7c7e)
12. [We're down with OP3](https://github.com/thechangelog/changelog.com/commit/023300a602c22cdf8dcfb350c36c462aa6a9d6fd)
13. [Fix a few outdated redirects](https://github.com/thechangelog/changelog.com/commit/248ea3b32c1cb71fe446496db002b8c60011e876)
14. [New feed keywords for Ship It!](https://github.com/thechangelog/changelog.com/commit/61a9f047298de609fdd07ad7eff85046a2aca676)
15. [Remove obsolete `<itunes:owner>` tag from feeds](https://github.com/thechangelog/changelog.com/commit/59d02ea9fe5a782cfc29d64168d22c58773f417d)
16. [Delete people from CM lists when deleting locally](https://github.com/thechangelog/changelog.com/commit/10fed565ebfe1ccc7cbf683d694af96d3fb27754)
17. [Only Weekly subscribers can submit news items](https://github.com/thechangelog/changelog.com/commit/0d855778f23d35830ae0bdcd49ebf547a1ff830d)
18. [Adjust source title for Changelog News episodes](https://github.com/thechangelog/changelog.com/commit/fe43856943ec300dcfa9c3f597280026f4289a6f)
19. [Fix bug in news item controller's new action](https://github.com/thechangelog/changelog.com/commit/25f58c78422da6fa7e9fbeb1c147dacaffa0ba61)
20. [Follow redirects when fetching remote temp files](https://github.com/thechangelog/changelog.com/commit/28df1d16bfdeaf3a445b3cd29dd79b52137a2fd7)
21. [Differentiate between audio public URL and direct URL](https://github.com/thechangelog/changelog.com/commit/dbe652537ad5621ca62b1df91d618e30137b146a)
22. [First draft of chapter support in the onsite player](https://github.com/thechangelog/changelog.com/commit/783ff0b176d3b00d8625b552b340688f6637180b)
23. [Don't autoplay after restoring onsite player state](https://github.com/thechangelog/changelog.com/commit/b2c92ffd7b1813dffe9b4ee62453573b59cd05c6)
24. [Close the onsite player when audio reaches its end](https://github.com/thechangelog/changelog.com/commit/e04f93bd85b982197f654936aac972999210c1c8)

I, @gerhard, am officially astonished 😲 with all the improvements that @jerodsanto shipped since **Kaizen 7**. If there is anyone that deserves to wear [a Kaizen t-shirt](https://merch.changelog.com/products/kaizen), that would be him! 👕

As I was going through / working on the above, I started thinking that maybe we should:
- sign & verify our commits - starting with @jerodsanto, continuing with @adamstac 
- improve our CI setup - PRs currently don't run tests 🤔
- start creating actual releases from these tags, starting with **Kaizen 8**
- double-check that our list of contributors is up-to-date

### 3/3. An open-ended discussion

The last improvement worth mentioning is this discussion. It was already mentioned at the top, but it's worth mentioning it again. If you want to ask questions before or after the 🎧 [Kaizen 8 episode](https://shipit.show/80), contribute to episode references, or simply dig into more details, it's all here, in this GitHub discussions. The best part is that it's an open-ended discussion because... [the show must go on 🎸](https://www.youtube.com/watch?v=t99KH0TR-J4)

---

https://changelog.com/topic/kaizen

<a href="https://changelog.com/topic/kaizen" target="_blank"><img width="1552" alt="image" src="https://user-images.githubusercontent.com/3342/200184476-cfc9ea56-01f3-4dc5-ac96-01de93c65db4.png"></a>

**Comment by adamstac** [comment]
Comment during the recording. Loving this!

**Comment by jerodsanto** [comment]
A few links for the show notes:

- [op3.dev](https://op3.dev)
- [Backstage with John Spurlock from OP3](https://changelog.com/backstage/25)
- [Kaizen t-shirts](https://merch.changelog.com/products/kaizen)
- [Jerod's Fly/Mastodon work](https://github.com/thechangelog/changelog.social)
